controllers/managedmcg_controller.go

```diff
@@ -488,67 +504,79 @@ func (r *ManagedMCGReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			prometheusRulesPredicates,
 		).
 		Complete(r)
+
+	return fmt.Errorf("error setting up ManagedMCG controller: %w", err)
 }
 
 func (r *ManagedMCGReconciler) lookupImages() error {
 	noobaaCoreImage, found := os.LookupEnv("NOOBAA_CORE_IMAGE")
 	if !found {
 		return fmt.Errorf("NOOBAA_CORE_IMAGE environment variable not set")
-	} else {
-		r.images.NooBaaCore = noobaaCoreImage
 	}
+	r.images.NooBaaCore = noobaaCoreImage
 	noobaaDBImage, found := os.LookupEnv("NOOBAA_DB_IMAGE")
 	if !found {
 		return fmt.Errorf("NOOBAA_DB_IMAGE environment variable not set")
-	} else {
-		r.images.NooBaaDB = noobaaDBImage
 	}
+	r.images.NooBaaDB = noobaaDBImage
+
 	return nil
 }
 
 func (r *ManagedMCGReconciler) get(obj client.Object) error {
 	key := client.ObjectKeyFromObject(obj)
-	return r.Client.Get(r.ctx, key, obj)
+	err := r.Client.Get(r.ctx, key, obj)
+
+	return fmt.Errorf("error getting %s: %w", key, err)
 }
 
 func (r *ManagedMCGReconciler) list(obj client.ObjectList) error {
 	listOptions := client.InNamespace(r.namespace)
-	return r.Client.List(r.ctx, obj, listOptions)
+	err := r.Client.List(r.ctx, obj, listOptions)
+
+	return fmt.Errorf("error listing %s: %w", obj, err)
 }
 
 func (r *ManagedMCGReconciler) update(obj client.Object) error {
-	return r.Client.Update(r.ctx, obj)
+	err := r.Client.Update(r.ctx, obj)
+
+	return fmt.Errorf("failed to update %s: %w", obj.GetName(), err)
 }
 
 func (r *ManagedMCGReconciler) delete(obj client.Object) error {
 	if err := r.Client.Delete(r.ctx, obj); err != nil && !errors.IsNotFound(err) {
-		return err
+		return fmt.Errorf("failed to delete %s: %w", obj.GetName(), err)
 	}
+
 	return nil
 }
 
 func (r *ManagedMCGReconciler) own(resource metav1.Object) error {
 	// Ensure ManagedMCG ownership on a resource
 	if err := ctrl.SetControllerReference(r.managedMCG, resource, r.Scheme); err != nil {
-		return err
+		return fmt.Errorf("failed to set controller reference on %s: %w", resource.GetName(), err)
 	}
+
 	return nil
 }
 
 func (r *ManagedMCGReconciler) getCSVByPrefix(name string) (*opv1a1.ClusterServiceVersion, error) {
 	csvList := opv1a1.ClusterServiceVersionList{}
 	if err := r.list(&csvList); err != nil {
-		return nil, fmt.Errorf("unable to list csv resources: %v", err)
+		return nil, fmt.Errorf("unable to list csv resources: %w", err)
 	}
-	var csv *opv1a1.ClusterServiceVersion = nil
+	var csv *opv1a1.ClusterServiceVersion
 	for _, candidate := range csvList.Items {
 		if strings.HasPrefix(candidate.Name, name) {
-			csv = &candidate
+			candidateCSV := candidate
```

Comment:
Dose this make sense?

Reply:
Fixed. Actually one of the linters had a problem with this due to the fact that the address will always be the same (only the value for the second for variable will change).

I've fixed this by using the first for key in the method, PTAL: https://stackoverflow.com/a/62486667.

A bit more on the explaination (copied from the first comment on the SO thread above):
```
Your for loop uses just 1 variable v and the value of v is updated by the for loop. But as it is just one variable it has just one address: &v is constant and does not change during the loop iteration. This is a typical problem in concurrent code as all created worker will work on the same memory location and thus either on the same value or on a value which is racy.
```